models/classes/evnetHandler/ResultTransmissionEventHandler/ResultTransmissionEventHandler.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\classes\evnetHandler\ResultTransmissionEventHandler;
+
+use oat\oatbox\service\ServiceManager;
+use oat\tao\model\service\InjectionAwareService;
+use oat\taoDelivery\model\execution\DeliveryServerService;
+use oat\taoQtiTest\models\classes\event\ResultTestVariablesTransmissionEvent;
+use oat\taoQtiTest\models\event\ResultItemVariablesTransmissionEvent;
+use taoQtiCommon_helpers_ResultTransmitter;
+
+class ResultTransmissionEventHandler extends InjectionAwareService implements Api\ResultTransmissionEventHandlerInterface
+{
+    /**
+     * @throws \taoQtiCommon_helpers_ResultTransmissionException
+     */
+    public function transmitResultItemVariable(ResultItemVariablesTransmissionEvent $event): void
+    {
+        $this->buildTransmitter($event->getDeliveryExecutionId())->transmitItemVariable(
+            $event->getVariables(),
+            $event->getTransmissionId(),
+            $event->getItemUri(),
+            $event->getTestUri()
+        );
+
+    }
+
+    /**
+     * @throws \taoQtiCommon_helpers_ResultTransmissionException
+     */
+    public function transmitResultTestVariable(ResultTestVariablesTransmissionEvent $event): void
+    {
+        $this->buildTransmitter($event->getDeliveryExecutionId())->transmitTestVariable(
+            $event->getVariables(),
+            $event->getTransmissionId(),
+            $event->getTestUri()
+        );
+    }
+
+    private function buildTransmitter($deliveryExecutionId): taoQtiCommon_helpers_ResultTransmitter
+    {
+        /** @var DeliveryServerService $deliveryServerService */
+        $deliveryServerService = ServiceManager::getServiceManager()->get(DeliveryServerService::SERVICE_ID);
```

Comment:
Would it be possible to use DI here? 

Even if no, you should access service manager like `$this->getServiceManager()`, which is implemented at `InjectionAwareService`